code/modules/antagonists/superheroes/buzzon/buzzon_equipment.dm

```diff
@@ -0,0 +1,294 @@
+	icon_state = "buzzon_sword"
+	inhand_icon_state = "buzzon"
+	worn_icon_state = "buzzon"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	slot_flags = ITEM_SLOT_BACK
+	block_chance = 10 //It's about agility, not about sheer force
+	///If this var is higher than world.time this means that the owner is still in the blocking stance and has a high chance to block the attack
+	var/parry_time = 0
+
+/obj/item/melee/beesword/buzzon/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(parry_time >= world.time)
+		parry_time = 0
+		final_block_chance += 85
+
+	switch(attack_type)
+		if(UNARMED_ATTACK)
+			final_block_chance += 30
+
+		if(PROJECTILE_ATTACK)
+			final_block_chance -= 40
+
+		if(LEAP_ATTACK)
+			if(isliving(hitby))
+				var/mob/living/victim = hitby
+				force = 20
+				attack_verb_continuous = list("penetrates", "pierces") //We pierce them in the flight.
+				attack_verb_simple = list("penetrate", "pierce")
+				attack(victim, owner)
+				force = initial(force)
+				attack_verb_continuous = initial(attack_verb_continuous)
+				attack_verb_simple = initial(attack_verb_simple)
+	. = ..()
+
+/obj/item/melee/beesword/buzzon/attack_secondary(mob/living/victim, mob/living/user, params)
+	user.changeNext_move(CLICK_CD_RANGE)
+	parry_time = world.time + CLICK_CD_RANGE
+	user.visible_message("<span class='warning'>[user] assumes blocking stance.</span>", "<span class='notice'>You assume blocking stance.</span>")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/melee/beesword/buzzon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(ishuman(hit_atom))
+		var/mob/living/carbon/human/victim = hit_atom
+		if(istype(victim.get_item_by_slot(ITEM_SLOT_OCLOTHING), /obj/item/clothing/suit/hooded/bee_costume/buzzon))
+			var/obj/item/clothing/suit/hooded/bee_costume/buzzon/suit = victim.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+			if(suit.linked_sword == src)
+				victim.put_in_active_hand(src)
+				victim.visible_message("<span class='warning'>[victim] catches [src] out of the air!</span>")
+				return
+	. = ..()
+
+/mob/living/simple_animal/hostile/bee/toxin_type
+	var/toxin_type = /datum/reagent/toxin
+
+/mob/living/simple_animal/hostile/bee/toxin_type/Initialize()
+	. = ..()
+	assign_reagent(GLOB.chemical_reagents_list[toxin_type])
+
+/mob/living/simple_animal/hostile/bee/toxin_type/cryo
+	toxin_type = /datum/reagent/consumable/frostoil
+
+/obj/item/grenade/spawnergrenade/buzzkill/non_toxic
+	spawner_type = /mob/living/simple_animal/hostile/bee
+	deliveryamt = 5
+
+/obj/item/clothing/glasses/hud/security/sunglasses/blue
+	icon_state = "sunhudmed"
+	glass_colour_type = /datum/client_colour/glass_colour/blue
+
+/obj/item/melee/beesword/buzzon/cryo
```

Comment:
ooooooooh, this is why you don't choose the cold option (or maybe this is yet another reason to pick it over the normal one, I'll have to do some testing)